### PR TITLE
Create review by user

### PR DESCRIPTION
**Pull request by Rownaknira**

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class BooksController < ApplicationController
   def index
+    @books = Book.all
```
Comment (noor-framgia) [inline app/controllers/books_controller.rb]: no need

**Comment by noor-framgia** [comment]
ok

**Comment by SSalekin** [comment]
ok

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -3,5 +3,6 @@ def index
   end
 
   def show
+    @reviews = Review.find_review_by_book @book.id
```
Comment (dieunb) [inline app/controllers/books_controller.rb]: why not `@book.reviews`

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -2,4 +2,19 @@ class ReviewsController < ApplicationController
   def index
     @activities = Review.review_acitvities
   end
+
+  def create
+    @review = Review.new review_params
```
Comment (dieunb) [inline app/controllers/reviews_controller.rb]: cancancan???

File: app/views/books/index.html.erb [hunk 4]
```diff
@@ -47,10 +47,10 @@
             <%= book.num_of_pages %>
           </td>
           <td>
-            <%= book.category.name %>
+            <%= book.category.name if book.category %>
```
Comment (linhnt) [inline app/views/books/index.html.erb]: use try

File: app/views/books/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <b><%= t ".book_author" %> : </b> <%= @book.author %>
+  </div>
+  <div class="panel-body">
+    <b><%= t ".book_publish_date" %> : </b> <%= @book.publish_date %>
+  </div>
+  <div class="panel-body">
+    <b><%= t ".book_num_of_pages" %> : </b> <%= @book.num_of_pages %>
+  </div>
+  <div class="panel-body">
+    <b><%= t ".book_category" %> : </b> <%= @book.category.name %>
```
Comment (linhnt) [inline app/views/books/show.html.erb]: book.category can be nil, right ?

Comment (Rownaknira) [inline app/views/books/show.html.erb]: here book.category is mandatory field

Comment (linhnt) [inline app/views/books/show.html.erb]: i asked because you check if category is present above.

File: app/models/review.rb [hunk 6]
```diff
@@ -5,6 +5,8 @@ class Review < ActiveRecord::Base
   belongs_to :book
   has_many :comments, dependent: :destroy
 
+  scope :find_review_by_book, ->(book_id) {where book_id: book_id}
```
Comment (dieunb) [inline app/models/review.rb]: remove it

File: app/views/books/show.html.erb (hunk 5 above)
Comment (Rownaknira) [inline app/views/books/show.html.erb]: sorry. I should be careful. I will be more careful next time

**Pull request merged**